Sources/WinAppDriver/WinAppDriver.swift

```diff
@@ -34,12 +40,59 @@ public class WinAppDriver: WebDriver {
         executablePath: String = defaultExecutablePath,
         ip: String = defaultIp,
         port: Int = defaultPort,
-        waitTime: TimeInterval? = defaultStartWaitTime) throws -> WinAppDriver {
-
+        waitTime: TimeInterval? = defaultStartWaitTime,
+        outputFile: String? = nil) throws -> WinAppDriver {
         let processTree: Win32ProcessTree
+        var childStdinHandle: HANDLE? = nil
         do {
-            processTree = try Win32ProcessTree(path: executablePath, args: [ ip, String(port) ])
+            var launchOptions = ProcessLaunchOptions()
+            if let outputFile = outputFile {
+                // Open the output file for writing to the child stdout.
+                var securityAttributes = SECURITY_ATTRIBUTES()
+                securityAttributes.nLength = DWORD(MemoryLayout<SECURITY_ATTRIBUTES>.size)
+                securityAttributes.bInheritHandle = true
+                launchOptions.stdoutHandle = try outputFile.withCString(encodedAs: UTF16.self) {
+                    outputFile throws in
+                    CreateFileW(
+                        UnsafeMutablePointer<WCHAR>(mutating: outputFile), DWORD(GENERIC_WRITE),
+                        DWORD(FILE_SHARE_READ), &securityAttributes,
+                        DWORD(OPEN_ALWAYS), DWORD(FILE_ATTRIBUTE_NORMAL), nil)
+                }
+                if launchOptions.stdoutHandle == INVALID_HANDLE_VALUE {
+                    // Failed to open the output file for writing.
+                    throw Win32Error.getLastError(apiName: "CreateFileW")
+                }
+
+                // Use the same handle for stderr.
+                launchOptions.stderrHandle = launchOptions.stdoutHandle
+
+                // WinAppDriver will close immediately if no stdin is provided so create a dummy
+                // pipe here to keep stdin open until the child process is closed.
+                var childReadInputHandle: HANDLE?
+                if !CreatePipe(&childReadInputHandle, &childStdinHandle, &securityAttributes, 0) {
+                    CloseHandle(launchOptions.stdoutHandle)
+                    throw Win32Error.getLastError(apiName: "CreatePipe")
+                }
+                launchOptions.stdinHandle = childReadInputHandle
+
+                // Also use the parent console to stop spurious new consoles from spawning.
+                launchOptions.spawnNewConsole = false
+            }
+
+            // Close our handles when the process has launched. The child process keeps a copy.
+            defer {
```

Comment:
This block needs to go before line 69, otherwise we can still leak the stdout handle if line 74 throws @Steelskin

Reply:
See above comment, we can't close the handles until after the process has launched.

Reply:
The defer block still achieves that if moved between lines 48 and 49